Fix inconsistent formatting logic of CallExpression and NewExpression (#18171)

harshasiddartha opened the pull request:
## Description

This PR fixes issue #18171 by making `CallExpression` and `NewExpression` format consistently when they contain conditional expressions or member expressions as arguments.

**Changes:**
- Updated `src/language-js/print/binaryish.js` to treat `NewExpression` the same as `CallExpression` for binary expression formatting (lines 88 and 113)
- Updated `src/language-js/print/ternary.js` and `src/language-js/print/ternary-old.js` to treat `NewExpression` the same as `CallExpression` in `shouldExtraIndentForConditionalExpression`

**Before this fix:**
```js
// CallExpression - properly indented
fn(
  arg1,
  arg2,
  condition1 &&
    condition2
    ? value1
    : value2,
);

// NewExpression - inconsistent indentation
new fn(
  arg1,
  arg2,
  condition1 &&
  condition2
    ? value1
    : value2,
);
```

**After this fix:**
```js
// Both now format consistently
fn(
  arg1,
  arg2,
  condition1 &&
    condition2
    ? value1
    : value2,
);

new fn(
  arg1,
  arg2,
  condition1 &&
    condition2
    ? value1
    : value2,
);
```

## Checklist

- [x] I've added tests to confirm my change works.
- [ ] (If changing the API or CLI) I've documented the changes I've made (in the `docs/` directory).
- [x] (If the change is user-facing) I've added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I've read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

Fixes #18171

fisker commented:
Did you do it with AI? Please try to revert some of them and make sure only necessary changes are made.

harshasiddartha commented:
Hi @fisker, thank you for the feedback! I've updated the PR to include only the minimal necessary changes to fix the issue.

**Changes made:**
- **Only one code change**: Added `&& grandparent.type !== "NewExpression"` check in `src/language-js/print/binaryish.js` (line 113) to treat `NewExpression` the same as `CallExpression` when determining indentation for binary expressions inside conditional expressions.

**Reverted unnecessary changes:**
- Removed changes to `ternary.js` and `ternary-old.js` (not needed for this fix)
- Removed the `parent.callee === node` check for `NewExpression` (not related to this issue)

The fix ensures that when a ternary expression containing binary expressions is passed as an argument to either `CallExpression` or `NewExpression`, both format consistently. The core issue was that the binary expression formatter checked for `CallExpression` as the grandparent but not `NewExpression`, causing inconsistent indentation.

All tests pass, including the new test case added for this issue.

fisker commented:
Please spend more time. This is not the correct solution.
